# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/QuotaManager.h
# Commit: dca9f407ea90
# Full Hash: dca9f407ea90a83ae23be6c48aab8ce3f0f6e23e
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-06-14 20:49:45
# Regressor Bug: 1883353
# File Overlap Count: 1
# Description:
#   Bug 1883353 - Require to always drop directory locks explicitly; r=dom-storage-reviewers,jari
#   
#   Until now, directory locks were dropped when the last strong reference was
#   removed or after calling Drop explicitly. The dependency on ref-counting makes
#   it less obvious when directory locks are dropped for real and it's also
# ==============================================================================

diff -r a82c4a895b9d -r dca9f407ea90 dom/quota/QuotaManager.h
--- a/dom/quota/QuotaManager.h	Wed Jun 12 15:12:34 2024 +0000
+++ b/dom/quota/QuotaManager.h	Fri Jun 14 07:21:24 2024 +0000
@@ -291,10 +291,9 @@
   // is resolved, there are no longer other strong references except the one
   // held by the resolve value itself. So it's up to client to add a new
   // reference in order to keep the lock alive.
-  // Unlocking is simply done by dropping all references to the lock object.
-  // In other words, protection which the lock represents dies with the lock
-  // object itself (Note that it's now possible to release directory locks
-  // sooner by calling newly added Drop method).
+  // Unlocking is simply done by calling lock object's Drop method. Unlocking
+  // must be always done explicitly before the lock object is destroyed (when
+  // the last strong reference is removed).
   RefPtr<ClientDirectoryLockPromise> OpenClientDirectory(
       const ClientMetadata& aClientMetadata,
       Maybe<RefPtr<ClientDirectoryLock>&> aPendingDirectoryLockOut = Nothing());